dnf/cli/option_parser.py

```diff
@@ -377,38 +366,45 @@ def get_usage(self):
                     usage += "%-25s %s\n" % (name, summary)
         return usage
 
-    def add_argument(self, *args, **kwargs):
-        # process options and arguments separately
+    def _add_command_options(self, command):
+        self.prog = "%s %s" % (self.prog, command._basecmd)
+        self.description = command.summary
+        self.command_positional_parser = argparse.ArgumentParser(self.prog, add_help=False)
+        self.command_positional_parser.print_usage = self.print_usage
+        self.command_positional_parser._positionals.title = None
+        self.command_group = self.add_argument_group(
+            'Command "{}" specific options'.format(command._basecmd))
+        self.command_group.add_argument = self.cmd_add_argument
+        command.set_argparser(self.command_group)
+
+    def cmd_add_argument(self, *args, **kwargs):
         if all([(arg[0] in self.prefix_chars) for arg in args]):
-            return super(OptionParser, self).add_argument(*args, **kwargs)
+            return argparse._ArgumentGroup.add_argument(self.command_group, *args, **kwargs)
```

Comment:
Private member though? Not so noice?

Reply:
Yes, I'm not proud of it...
Let me think about it once more (although work with python's argparse module is quite tricky). I'm moving PR to WIP meanwhile.

Reply:
This one was quite easy to remove.